test/python3/libdnf/rpm/test_solv_query.py

```diff
@@ -51,6 +51,61 @@ def test_size(self):
         query = libdnf.rpm.SolvQuery(self.sack)
         self.assertEqual(query.size(), 291)
 
+    def test_iterate_packageset(self):
+        query = libdnf.rpm.SolvQuery(self.sack)
+
+        # Iterates over helping reference "pset".
+        pset = query.get_package_set()
+        prev_id = 1
+        for pkg in pset:
+            id = pkg.get_id().id
+            self.assertEqual(id, prev_id + 1)
+            prev_id = id
+        self.assertEqual(prev_id + 1, self.sack.get_nsolvables())
+
+        # Similar to above, but longer notation.
+        pset_iterator = iter(pset)
+        prev_id = 1
+        while True:
+            try:
+                pkg = next(pset_iterator)
+                id = pkg.get_id().id
+                self.assertEqual(id, prev_id + 1)
+                prev_id = id
+            except StopIteration:
+                break
+        self.assertEqual(prev_id + 1, self.sack.get_nsolvables())
+
+    def test_iterate_packageset2(self):
+        # Tests the iteration of an unreferenced PackageSet object. The iterator must hold a reference
+        # to the iterated object, otherwise the gargabe collector can remove the object.
+
+        query = libdnf.rpm.SolvQuery(self.sack)
+
+        # Iterates directly over "get_package_set()" result. No helping reference.
+        prev_id = 1
+        for pkg in query.get_package_set():
+            id = pkg.get_id().id
+            self.assertEqual(id, prev_id + 1)
+            prev_id = id
+        self.assertEqual(prev_id + 1, self.sack.get_nsolvables())
```

Comment:
I would prefer `self.assertGreaterEqual(prev_id, self.sack.get_nsolvables())`. It will not make any problem when we will add advisoriesy into test data. Then the number of solvables will not end with the latest RPM package but with a patch.

Reply:
If advisory is at the and of the pool, then `prev_id` is lesser than `self.sack.get_nsolvables()` -> `self.assertLess(prev_id, self.sack.get_nsolvables())`.

I replaced assert by two asserts (I gave back assert with `query.size()`)
```
self.assertLess(prev_id, self.sack.get_nsolvables())
self.assertGreater(prev_id, query.size())
```